neurom/core/_neuron.py

```diff
@@ -152,6 +154,51 @@ def graft_neuron(root_section):
     return Neuron(soma=Soma(root_section.points[:1]), neurites=[Neurite(root_section)])
 
 
+def iter_positions(morph, neurite_filter, sample_distance):
+    """
+    Iterator for positions in space of points every <sample_distance> um
+
+    Args:
+        morph (neurom.FstNeuron): morphology
+        neurite_type: (neurom.NeuriteType): filter by this type of neurite
+        sample_distance (int in um): points sampling distance
+
+    Yields:
+        sampled points for the neurites. each point is a (3,) numpy array
+
+    Assumptions:
+        segment linearity: curvature between the start and end of segments
+            is negligible
+    """
+    # map of section to its remaining offset
+    section_offsets = {}
+
+    for section in iter_sections(morph, neurite_filter=neurite_filter):
+        if section.parent is None:
+            parent_section_offset = 0
+        else:
+            parent_section_offset = section_offsets[section.parent.id]
+        segment_offset = parent_section_offset
+        for segment in iter_segments(section):
+            segment_len = morphmath.segment_length(segment)
+            if segment_offset + segment_len < sample_distance:
+                segment_offset += segment_len
+            elif segment_offset + segment_len == sample_distance:
+                yield segment[1][COLS.XYZ]
+                segment_offset = 0
+            else:
+                offsets = np.arange(sample_distance - segment_offset, segment_len, sample_distance)
+                for offset in offsets:
+                    yield morphmath.linear_interpolate(*segment, offset / segment_len)
```

Comment:
I think since it isn't needed for anything at present I'd prefer to leave adding other sampled features for another time or another person so that we can move on with these usecases (and so I can focus on starting my PhD).

Reply:
The problem there is multiple API's will exist to do something similar, and then we have a deprecation problem.

>  (and so I can focus on starting my PhD).

I'm trying to keep that in mind, and, again, I appreciate the contribution.

Reply:
Go start your PhD, Hugo! I can add this in later today so we'll have more features available with this functionality!